src/components/nav_selector/index.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import { Animated, GestureResponderEvent, PressableProps } from 'react-native'
+
+import * as Styled from './styles'
+import TabSelector from './tab_selector'
+
+export interface Props extends Omit<PressableProps, 'onPress'> {
+  leading: { label: string; icon: JSX.Element }
+  trailing: { label: string; icon: JSX.Element }
+  defaultSelected?: string
+  onPress?: (e: GestureResponderEvent, value: string) => void
+}
+
+const NavSelector = (props: Props): JSX.Element => {
+  const { leading, trailing, defaultSelected, onPress } = props
+
+  const [selected, setSelected] = React.useState(defaultSelected ?? leading.label)
+
+  const [leadingWidth, setLeadingWidth] = React.useState(1)
```

Comment:
cant we have only one state here for width? and set on onPress on the tabSelector